File: Programming Basics/While Loop - Exercises/04.Walking.cs
using System;
class Program
{
    static void Main(string[] args)
    {
        string steps = null;
        int walkedSteps = 0;

        while (walkedSteps < 10000)
        {
            steps = Console.ReadLine();

            if (steps == "Going home")
            {
                walkedSteps += int.Parse(Console.ReadLine());
                break;
            }
            if (walkedSteps >= 10000)
            {
                GoalReached(walkedSteps);
                System.Environment.Exit(0);
            }
            walkedSteps += int.Parse(steps);
        }
        if (walkedSteps >= 10000)
        {
            GoalReached(walkedSteps);
        }
        else
        {
            Console.WriteLine($"{10000 - walkedSteps} more steps to reach goal.");
        }
    }
    static void GoalReached(int walkedSteps)
    {
        Console.WriteLine("Goal reached! Good job!");
        Console.WriteLine($"{walkedSteps - 10000} steps over the goal!");
    }
}

File: Programming Basics - Python/While Loop - Exercises/04.Walking.py
steps = input()

goal = 10_000

steps_for_today = 0

while steps != "Going home":
    steps_for_today += int(steps)

    if goal <= steps_for_today:
        print("Goal reached! Good job!")
        print(f"{steps_for_today - goal} steps over the goal!")
        exit()

    steps = input()

steps_for_today += int(input())

if goal <= steps_for_today:
    print("Goal reached! Good job!")
    print(f"{steps_for_today - goal} steps over the goal!")

else:
    print(f"{goal - steps_for_today} more steps to reach goal.")